src/components/datetime/datetime.ts

```diff
@@ -619,28 +619,14 @@ export class DateTime extends Ion implements AfterContentInit, ControlValueAcces
       }
     }
 
-    // default to assuming this month has 31 days
```

Comment:
I moved the code to be closer to where it is used; to make it easier to understand and encourage refactoring.  The moved block could eventually be scoped inside of `if (dayCol)`.

If you think it is better where it was originally located, I will move it back.